src/CPE/CMakeLists.txt

```diff
@@ -0,0 +1,13 @@
+file(GLOB_RECURSE CPE_SOURCES "*.c")
+file(GLOB_RECURSE CPE_HEADERS "*.h")
+file(GLOB_RECURSE PUBLIC_HEADERS "public/*.h")
+
+add_library(cpe_object OBJECT ${CPE_SOURCES} ${CPE_HEADERS})
+add_library(cpe $<TARGET_OBJECTS:cpe_object>)
+target_link_libraries(cpe common)
+
+set_target_properties(cpe_object PROPERTIES
+    CXX_VISIBILITY_PRESET hidden	# Make global variables and functions HIDDEN by default.
+    POSITION_INDEPENDENT_CODE ON	# Compile this object code position independent.
+)
```

Comment:
you set this in all the files, perhaps make a macro/function and call it, it can set properties for every openscap target

Reply:
I was looking into this and I am pretty sure you are using features here that are not available in cmake 2.6 or 2.8. According to docs CXX_VISIBILITY_PRESET is in 3.0.2 and onwards. Or maybe I am reading it wrong?

Reply:
@mpreisler Removing this leads to a large amount of errors of this format and it doesn't build.
```
/usr/bin/ld: common/CMakeFiles/common_object.dir/xml_iterate.c.o: relocation R_X86_64_32 against `.rodata' can not be used when making a shared object; recompile with -fPIC
```
That looks like a reason to have it there.

I will introduce a function to reduce copy-pasting.